### PR TITLE
Update the Interactivity API files to include latest changes

### PR DESCRIPTION
**Pull request by DAreRodz**
## What?

Updates all files related to the Interactivity API.

Tracking issue: https://github.com/woocommerce/woocommerce/issues/42486

## Why?

To maintain it updated, and thus be able to start using it in blocks (e.g. the "Add to cart" block).

## How?

The JS runtime code comes from [WordPress/block-interactivity-experiments](https://github.com/WordPress/block-interactivity-experiments), but the PHP one comes from [WordPress/gutenberg](https://github.com/WordPress/gutenberg) (more precisely, from the `interactivity` branch).

All prefixes related to `wp` and `gutenberg` were replaced with `wc` and `woocommerce` ones.

There is also some custom code to register the Interactivity API runtime script.

## Testing instructions

You can check the Simple Price Filter block using the updated Interactivity API in the next PR:
- https://github.com/woocommerce/woocommerce-blocks/pull/9926

* [x] Do not include in the Testing Notes

### CODE REVIEW

**Comment by luisherranz** [comment]
I don't think the server-side rendering implementation is ready to be used in production.

**Comment by DAreRodz** [comment]
> I don't think the server-side rendering implementation is ready to be used in production.

I removed all files related to directive processing during SSR-ing. The `store`, the runtime script registering, and the code including the CSN meta tag in `<head>` are the only remaining parts.

**Comment by luisherranz** [comment]
Thanks David! 🙂

**Comment by DAreRodz** [comment]
I think this is ready for review again. 🙂

**Comment by DAreRodz** [comment]
I almost forgot to replicate the changes @danielwrobert did to [adjust JSON escaping for the Store script](https://github.com/woocommerce/woocommerce-blocks/pull/9709). 😄

Added in https://github.com/woocommerce/woocommerce-blocks/pull/9924/commits/e0a3e8d37cbb487f70d42c3e9455cedb8bbeb184.

**Review by luisherranz — APPROVED** [review]
LGTM! Thanks David 🎉

**Comment by luisherranz** [comment]
@gigitux could you please also take a look at it?

**Comment by luisherranz** [comment]
> I almost forgot to replicate the changes [@danielwrobert](https://github.com/danielwrobert) did to [adjust JSON escaping for the Store script](https://github.com/woocommerce/woocommerce-blocks/pull/9709). 😄

Oh, @DAreRodz, would you mind doing that change in Gutenberg as well? Thanks!! 🙏

**Comment by DAreRodz** [comment]
> @gigitux could you please also take a look at it?

Luigi, feel free to ping someone else if you feel they need to review these changes. 😊

File: bin/webpack-configs.js [hunk 1]
```diff
@@ -785,11 +785,17 @@ const getInteractivityAPIConfig = ( options = {} ) => {
 	const { alias, resolvePlugins = [] } = options;
 	return {
 		entry: {
-			runtime: './assets/js/interactivity',
+			'wc-interactivity': './assets/js/interactivity',
 		},
 		output: {
-			filename: 'woo-directives-[name].js',
```
Comment (gigitux) [inline bin/webpack-configs.js:791]: Is there a specific reason for this change?

**Review by gigitux — APPROVED** [review]
LGTM! I'm just curious about a change, but let's 🚢  it!

File: bin/webpack-configs.js (hunk 1 above)
Comment (DAreRodz) [reply bin/webpack-configs.js:791]: It's just that I changed the name to make it consistent with the format used in other WooCommerce scripts in the repository. Nothing important. 😄

Note that there's still a WooCommerce prefix in that file; the name of the output file has turned from `woo-directives-runtime.js` to `wc-interactivity.js`.

Comment (DAreRodz) [reply bin/webpack-configs.js:791]: I also removed the `vendors` chunk, so I saw it unnecessary to differentiate between `runtime` and `vendors`.

**Comment by DAreRodz** [comment]
Thanks, folks, for the reviews. PR merged. 🚀

**Pull request merged**